### DISCUSSION TITLE
issue with isolates and typedResult

### DISCUSSION THREAD

**Discussion by BananaMasterz** [simolus3/drift#2420] [Q&A] [answered]
Just updated to flutter 3.10 and now when I try to return a `List<TypedResult>` from a `computeWithDatabase` function I get the following error:

```
[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: Invalid argument:  <- Instance of 'DriftClient' (from package:drift/src/remote/client_impl.dart)
 <- Instance of '_RemoteStreamQueryStore' (from package:drift/src/remote/client_impl.dart)
 <- Instance of 'DatabaseConnection' (from package:drift/src/runtime/api/runtime_api.dart)
 <- Instance of 'MyDatabase' (from package:my_app/database/database.dart)
 <- Instance of 'QueryRow' (from package:drift/src/runtime/query_builder/query_builder.dart)
 <- Instance of 'TypedResult' (from package:drift/src/runtime/query_builder/query_builder.dart)
 <- _List len:3 (from dart:core)
 <- Instance(length:1) of '_GrowableList' (from dart:core)
 <- _List len:1 (from dart:core)
is unsendable object (see restrictions listed at`SendPort.send()` documentation for more information): Instance of '_AsyncCompleter<ServerInfo>'
#0      Isolate._exit (dart:isolate-patch/isolate_patch.dart:647:75)
#1      Isolate.exit (dart:isolate-patch/isolate_patch.dart:650:5)
#2      _RemoteRunner._run (dart:isolate:1031:13)
<asynchronous suspension>
```

did something change that broke something?

**Comment by simolus3** [comment]
Thanks for the report! But I don't really understand why you'd send a `TypedResult` over a send port? Do you have a specific query that is expensive? If so, why not use drift's builtin isolate support or `NativeDatabase.createInBackground` and just use a regular query? Then you'd get the same thing but drift would serialize the results at a lower level.

A `TypedResult`/`QueryRow` needs to know how to interpret raw SQL data as Dart types, which can depend on database options (hence the reference to `MyDatabase` in there). I thought this would be fine since I didn't imagine a need to transfer those classes, but if you have a compelling use case that I'm not aware of we can revisit this assumption.

**Reply by BananaMasterz** [reply]
Hmm I'm using `computeWithDatabase`, isn't that drift's built-in isolate support? The error says send port which I assumed is what drift's built-in isolate support uses. I may be doing something wrong here. I'll look more into it.

**Reply by simolus3** [reply, marked as answer]
Using `computeWithDatabase` was designed for expensive work units that need access to the database (basically, sharing a drift database + `Isolate.run`).
If you're opening your database with `NativeDatabase.createInBackground`, drift uses an internal background isolate to run your queries already. So there's no need to wrap expensive queries with `computeWithDatabase`, if that's what you were trying to do.

**Reply by BananaMasterz** [reply]
Oh I didn't know that... Was this introduced at the same time as `computeWithDatabase` or has it always been like that?

**Reply by simolus3** [reply]
Initially you had to spawn the `DriftIsolate`s yourself to benefit from them, `createInBackground` and `computeWithDatabase` have been introduced much later (and IIRC around the same time) to make using drift isolates much easier to use.
